backend/src/src-licenses/src/main/java/org/eclipse/sw360/licenses/tools/OSADLObligationConnector.java

```diff
@@ -0,0 +1,322 @@
+			if (responseCode != HttpURLConnection.HTTP_OK) {
+				log.error("Could not open OSADL License URL: " + obligationURL);
+				log.error("HTTP Response Code: " + responseCode);
+				return null;
+			}
+			BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
+			StringBuffer buffer = new StringBuffer();
+			String inputLine;
+			while ((inputLine = in.readLine()) != null) {
+				buffer.append(inputLine);
+				buffer.append(System.lineSeparator());
+			}
+			in.close();
+			return buffer.toString();
+		} catch (IOException e) {
+			log.error("Could not get OSADL License for: " + licenseId);
+			return null;
+		}
+    }
+
+	@Override
+	public JSONObject parseText(String obligationText) {
+        String lines = obligationText;
+        String[] arraylines = lines.split("\n");
+		List<String> jsonText = setLinePath(setLineLevel(arraylines));
+		return buildTreeObject(jsonText);
+	}
+
+	public static Optional<Obligation> get(String licenseId, User user) {
+		OSADLObligationConnector osadlConnector = new OSADLObligationConnector();
+		String obligationText = osadlConnector.getText(licenseId);
+		if (obligationText == null) {
+			return Optional.empty();
+		}
+
+		Obligation obligation = new Obligation();
+		obligation.setId(TMP_OBLIGATION_ID_PREFIX + UUID.randomUUID().toString());
+		obligation.setText(obligationText);
+		obligation.setTitle(licenseId);
+		obligation.setObligationLevel(ObligationLevel.LICENSE_OBLIGATION);
+		obligation.setObligationType(ObligationType.OBLIGATION);
+		obligation.setDevelopment(false);
+		obligation.setDistribution(false);
+		obligation.addToWhitelist(user.getDepartment());
+		obligation.setExternalIds(Collections.singletonMap(EXTERNAL_ID_OSADL, licenseId));
+		return Optional.of(obligation);
+	}
+
+	private List<String> setLineLevel(String[] arraylines) {
+		List<String> refinedLines = new ArrayList<>();
+		for (int i = 0; i < arraylines.length; i++) {
+			if (arraylines[i].isEmpty()) {
+				continue;
+			}
+			int currentLevel = getLevel(arraylines[i]);
+			String lineWithLevel = "{ 'id': '" + i + "', 'text': '" + arraylines[i].replace("\t","") + "', 'level': '" + currentLevel + "', 'path': '-1'}";
+			refinedLines.add(lineWithLevel);
+		}
+		return refinedLines;
+	}
+
+	private int getLevel(String line) {
+		return line.length() - line.replace("\t","").length();
+	}
+
+	private List<String> setLinePath(List<String> lines) {
+		List<String> refinedLines = new ArrayList<>();
+		int i = 0;
+		for (String line : lines) {
+			String parentLinePath = "-1";
+			try {
+				JSONObject currentLine = JSONFactoryUtil.createJSONObject(line);
+				int parentLineId = getParentLineId(currentLine, lines);
+				if (parentLineId >= 0) {
+					try {
+						JSONObject parentLine = JSONFactoryUtil.createJSONObject(lines.get(parentLineId));
+						parentLinePath = parentLine.get("path").toString();
+					} catch (Exception e) {
+						//TODO: handle exception
+					}
```

Comment:
Please handle the exception in the catch block above and in the below catch blocks.
Indentation in line 208 before {